### ISSUE TITLE
NAS搜到的模型retrain之后无法正常做reference

### ISSUE THREAD

**Issue by marsggbo** [microsoft/nni#2869]
我用DARTS算法搜到一个模型并retrain，retrain结果显示结果测试集结果有93% accuracy，可是当我load模型只跑一轮测试集后，accuracy却只有36.3%。我试过不load模型参数，accuracy也是30%左右。这可能是哪些地方有问题呢？

还有个疑问是存模型的时候存的是supernet参数，还是子模型参数？

**Comment by ultmaster** [comment]
Hi DARTS 的流程是先 train 一个 supernet 再 train 子模型。你指的存模型是第一阶段的存模型吗？如果是的话存的是 supernet

**Comment by marsggbo** [comment]
> Hi DARTS 的流程是先 train 一个 supernet 再 train 子模型。你指的存模型是第一阶段的存模型吗？如果是的话存的是 supernet

我存的是retrain阶段的模型，但是在重新load跑一遍test set后准确率就完全不对了

**Comment by ultmaster** [comment]
你确认过你真的 load checkpoint 了嘛？load 的时候有没有出 error？apply fixed architecture 了吗？

**Comment by marsggbo** [comment]
> 你确认过你真的 load checkpoint 了嘛？load 的时候有没有出 error？apply fixed architecture 了吗？

我对比过load前后最后全连接层的参数，确定参数发生了变化，load过程也没有报错。apply了。

我试过model.train()和model.eval() 两种情况，发现model.train()结果会稍微好一些，不过也比保存模型时记录的结果差了很多

**Comment by ultmaster** [comment]
你存模型的时候是怎么存的？会不会有一些层（比如 BN 之类的）被遗漏了？

另外，需要先 apply_fixed_architecture 再 load checkpoint

**Comment by kvartet** [comment]
> 你存模型的时候是怎么存的？会不会有一些层（比如 BN 之类的）被遗漏了？
> 
> 另外，需要先 apply_fixed_architecture 再 load checkpoint

@marsggbo Any updates? Thank you.
